### PR TITLE
Fix Replace actor bugfix + fix test suites

### PR DESCRIPTION
**Pull request by Lqp1**
This contains:
* simplify of healthiness logic
* Allow marathon to start a new task, directly after sending killing order
* remove unused method
* Fixed RestartStrategy for some corner cases
* adds a test in RestartStrategy to check the end of a deployment
* add fixtures in TaskReplaceActorTest
* fix tests of TaskReplaceActor (some tests changed because RestartStrategy is slightly different now)

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala [hunk 1]
```diff
@@ -255,9 +244,10 @@ object TaskReplaceActor extends StrictLogging {
     def canStartNewInstances: Boolean = minHealthy < maxCapacity || consideredHealthyInstancesCount - nrToKillImmediately < maxCapacity
     assume(canStartNewInstances, "must be able to start new instances")
 
-    val leftCapacity = math.max(0, maxCapacity - totalInstancesRunning)
+    val leftCapacity = math.max(0, maxCapacity - totalInstancesRunning + nrToKillImmediately)
```
Comment (fquesnel) [inline src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala]: Would it be an issue to remove this? Will the deployment take longer to complete? Will some tests fail?

Comment (Lqp1) [reply src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala]: Now that CI is fully working I'll have a look if I can remove this

**Review by fquesnel — APPROVED** [review]
Lgtm

**Comment by Lqp1** [comment]
Squashed and Merged manually into https://github.com/criteo-forks/marathon/commits/marathon-replace-bugfix
